### ISSUE TITLE
[BUG] some unit tests fail on GPUs

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#2230]
### Bug summary

Although all tests pass on CPUs, some tests fail on GPUs.

### DeePMD-kit Version

devel

### TensorFlow Version

2.9.0

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

These two tests fail on 3080Ti but pass on 2080Ti
```
self = <test_tabulate.TestDPTabulate testMethod=test_op_tanh>

    def test_op_tanh(self):
        w=tf.constant([[0.1,0.2,0.3,0.4],[0.5,0.6,0.7,0.8],[0.9,1,1.1,1.2]],dtype='double')
        x=tf.constant([[0.1,0.2,0.3],[0.4,0.5,0.6],[0.7,0.8,0.9],[1.0,1.1,1.2]],dtype='double')
        b=tf.constant([[0.1],[0.2],[0.3],[0.4]],dtype='double')
        xbar = tf.matmul(x, w) + b
        y=tf.nn.tanh(xbar)
        dy = op_module.unaggregated_dy_dx_s(y, w, xbar, tf.constant(1))
        dy_array = tf.Session().run(dy)
        answer = np.array([[8.008666403121351973e-02, 1.513925729426658651e-01, 2.134733287761668430e-01, 2.661983049806041501e-01],
                           [4.010658815015744061e-02, 6.306476628799793926e-02, 7.332167904608145881e-02, 7.494218676568849269e-02],
                           [1.561705624394135218e-02, 1.994112926507514427e-02, 1.887519955881525671e-02, 1.576442161040989692e-02],
                           [5.492686739421748753e-03, 5.754985286040992763e-03, 4.493113544969218158e-03, 3.107638130764600777e-03]])
    
        places = 18
>       np.testing.assert_almost_equal(dy_array, answer, places)
E       AssertionError: 
E       Arrays are not almost equal to 18 decimals
E       
E       Mismatched elements: 10 / 16 (62.5%)
E       Max absolute difference: 2.77555756e-17
E       Max relative difference: 3.6283834e-15
E        x: array([[0.08008666403121352 , 0.15139257294266587 , 0.21347332877616684 ,
E               0.26619830498060415 ],
E              [0.04010658815015745 , 0.06306476628799794 , 0.07332167904608143 ,...
E        y: array([[0.08008666403121352 , 0.15139257294266587 , 0.21347332877616684 ,
E               0.26619830498060415 ],
E              [0.04010658815015744 , 0.06306476628799794 , 0.07332167904608146 ,...
```

```
self = <test_tabulate.TestDPTabulate testMethod=test_op_gelu>

    def test_op_gelu(self):
        w = tf.constant([[0.1, 0.2, 0.3, 0.4], [0.5, 0.6, 0.7, 0.8], [
                        0.9, 1, 1.1, 1.2]], dtype='double')
        x = tf.constant([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6], [
                        0.7, 0.8, 0.9], [1.0, 1.1, 1.2]], dtype='double')
        b = tf.constant([[0.1], [0.2], [0.3], [0.4]], dtype='double')
        xbar = tf.matmul(x, w) + b
        y = gelu(xbar)
        dy = op_module.unaggregated_dy_dx_s(y, w, xbar, tf.constant(2))
        dy_array = tf.Session().run(dy)
        answer = np.array([[8.549286163555620821e-02, 1.782905778685600906e-01, 2.776474599997448833e-01, 3.827650237273348965e-01],
                           [1.089906023807040714e-01, 2.230820937721638697e-01, 3.381867859682909927e-01, 4.513008399758057232e-01],
                           [1.124254240556722684e-01, 2.209918074710395253e-01, 3.238894323148118759e-01, 4.220357318198978414e-01],
                           [1.072173273655498138e-01, 2.082159073100979807e-01, 3.059816075270163083e-01, 4.032981557798429595e-01]])
    
        places = 18
>       np.testing.assert_almost_equal(dy_array, answer, places)
E       AssertionError: 
E       Arrays are not almost equal to 18 decimals
E       
E       Mismatched elements: 3 / 16 (18.8%)
E       Max absolute difference: 5.55111512e-17
E       Max relative difference: 1.99933942e-16
E        x: array([[0.08549286163555621, 0.1782905778685601 , 0.27764745999974494,
E               0.38276502372733495],
E              [0.10899060238070407, 0.22308209377216387, 0.338186785968291  ,...
E        y: array([[0.08549286163555621, 0.1782905778685601 , 0.2776474599997449 ,
E               0.3827650237273349 ],
E              [0.10899060238070407, 0.22308209377216387, 0.338186785968291  ,...
```

These two tests fail on both 2080Ti and 3080Ti
```
>       np.testing.assert_almost_equal(model_dout[0:10], ref_dout, places)
E       AssertionError: 
E       Arrays are not almost equal to 10 decimals
E       
E       Mismatched elements: 4 / 10 (40%)
E       Max absolute difference: 0.00024414
E       Max relative difference: 0.01818182
E        x: array([ 0.0065917969,  0.0076904297, -0.0256347656,  0.0081787109,
E               0.0091552734, -0.0301513672,  0.0096435547,  0.0122070312,
E               0.0986328125, -0.0317382812])
E        y: array([ 0.0067138672,  0.0078125   , -0.0258789062,  0.0081787109,
E               0.0091552734, -0.0302734375,  0.0096435547,  0.0122070312,
E               0.0986328125, -0.0317382812])

test_nvnmd_se_a.py:126: AssertionError
```

```
>       np.testing.assert_almost_equal(data['G_t0_t0'], data2['G_t0_t0'])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 27703 / 65536 (42.3%)
E       Max absolute difference: 0.00585938
E       Max relative difference: 3.
E        x: array([[[-8.6816406e-01, -2.6230469e+00,  2.1875000e+00, ...,
E                -9.8046875e-01, -2.5800781e+00,  1.2275391e+00],
E               [-8.6816406e-01, -2.6240234e+00,  2.1875000e+00, ...,...
E        y: array([[[-8.6816406e-01, -2.6230469e+00,  2.1875000e+00, ...,
E                -9.8046875e-01, -2.5800781e+00,  1.2275391e+00],
E               [-8.6718750e-01, -2.6230469e+00,  2.1875000e+00, ...,...

test_nvnmd_entrypoints.py:94: AssertionError
```

### Steps to Reproduce

Run tests on the 3080Ti

```sh
pytest source/tests
```

**Comment by wanghan-iapcm** [comment]
The first two test cases require to many matching decimals. 

The last two cases are from nvnmd, are they expected to pass on GPU ?  @LiuGroupHNU

**Comment by njzjz** [comment]
I cannot reproduce the first two cases with the latest codes. The reason is unclear. There may be some randomness.

I confirm the last two cases can be fixed by #2225.
